src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -285,42 +285,89 @@ public FeedbackSessionQuestionsBundle getFeedbackSessionQuestionsForInstructor(
-            // instructor can only see students in allowed sections for him/her
-            if (question.recipientType.equals(FeedbackParticipantType.STUDENTS)) {
-                Iterator<Map.Entry<String, String>> iter = recipients.entrySet().iterator();
-                while (iter.hasNext()) {
-                    Map.Entry<String, String> studentEntry = iter.next();
-                    StudentAttributes student = studentsLogic.getStudentForEmail(courseId, studentEntry.getKey());
-                    if (!instructor.isAllowedForPrivilege(student.section, 
-                            fsa.feedbackSessionName, Const.ParamsNames.INSTRUCTOR_PERMISSION_SUBMIT_SESSION_IN_SECTIONS)) {
-                        iter.remove();
-                    }
+            updateBundleAndRecipientListWithResponsesForInstructor(courseId, 
+                    userEmail, fsa, instructor, bundle, recipientList,
+                    question, instructorGiver, studentGiver);
+        }
+
+        return new FeedbackSessionQuestionsBundle(fsa, bundle, recipientList);
+    }
+    
+    public FeedbackSessionQuestionsBundle getFeedbackSessionQuestionsForInstructor(
+            String feedbackSessionName, String courseId, String feedbackQuestionId, String userEmail)
+            throws EntityDoesNotExistException {
+
+        FeedbackSessionAttributes fsa = fsDb.getFeedbackSession(
+                courseId, feedbackSessionName);
+
+        if (fsa == null) {
+            throw new EntityDoesNotExistException(
+                    "Trying to get a feedback session that does not exist.");
+        }
+
+        InstructorAttributes instructor = instructorsLogic.getInstructorForEmail(courseId, userEmail);
+        Map<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>> bundle
+            = new HashMap<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>>(); 
+        Map<String, Map<String,String>> recipientList
+            = new HashMap<String, Map<String,String>>();
+        
+        
+        FeedbackQuestionAttributes question = fqLogic.getFeedbackQuestion(feedbackQuestionId);
+        
+        InstructorAttributes instructorGiver = instructor;
+        StudentAttributes studentGiver = null;
+
+        updateBundleAndRecipientListWithResponsesForInstructor(courseId,
+                userEmail, fsa, instructor, bundle, recipientList,
+                question, instructorGiver, studentGiver);
+    
+
+        return new FeedbackSessionQuestionsBundle(fsa, bundle, recipientList);
+    }
+
+    private void updateBundleAndRecipientListWithResponsesForInstructor(
+            String courseId,
+            String userEmail,
+            FeedbackSessionAttributes fsa,
+            InstructorAttributes instructor,
+            Map<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>> bundle,
+            Map<String, Map<String, String>> recipientList,
```

Comment:
The variable name / type does not make sense here, but it is used in many places. Create a new issue to refactor?

Map<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>> bundle
Map<String, Map<String, String>> recipientList